### PR TITLE
Fixes thermomachines being able to run without power

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

Ports tgstation/tgstation#56261

## Why It's Good For The Game

it's a fix

## Changelog
:cl: Nari Harimoto
fix: fixed thermomachines still working even without power, apc off, in space etc...
/:cl:

### CODE REVIEW

**Review by utilbird — COMMENTED** [review]
`is_operational` is a proc, not a variable

**Comment by jupyterkat** [comment]
i can see the lints y'know

**Comment by utilbird** [comment]
I didn't even check the lints, was trying to help :p

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 1]
```diff
@@ -110,8 +111,10 @@
 
 /obj/machinery/atmospherics/components/unary/thermomachine/process_atmos()
 	..()
-	if(!on || !nodes[1])
+	if(!(is_operational()) || !on || !nodes[1])  //if it has no power or its switched off, dont process atmos
 		return
+	else if(is_operational() && was_on == TRUE)  //if it was switched on before it turned off due to no power, turn the machine back on
```
Comment (utilbird) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]:
```suggestion
	else if(is_operational() && was_on)  //if it was switched on before it turned off due to no power, turn the machine back on
```

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 2]
```diff
@@ -110,8 +111,10 @@
 
 /obj/machinery/atmospherics/components/unary/thermomachine/process_atmos()
 	..()
-	if(!on || !nodes[1])
+	if(!(is_operational()) || !on || !nodes[1])  //if it has no power or its switched off, dont process atmos
```
Comment (PowerfulBacon) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]:
```suggestion
	if(!is_operational() || !on || !nodes[1])  //if it has no power or its switched off, dont process atmos
```

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 3]
```diff
@@ -110,8 +111,10 @@
...
 		return
+	else if(is_operational() && was_on)  //if it was switched on before it turned off due to no power, turn the machine back on
+		on = TRUE
```
Comment (PowerfulBacon) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]: If it was switched off due to having no power on = FALSE, meaning this line will never run

**Review by utilbird — APPROVED** [review]

**Pull request merged**